### PR TITLE
Add Board support for Geekble ESP32-C3 Custom Module

### PR DESCRIPTION
**Pull request by SooDragon**
## Description of Change

Add board support for Geekble ESP32C3 Module

## Tests scenarios

I have tested with Arduino IDE, Version 2.3.1 and 1.8.19. Verified by compiling, uploading, testing using IO Examples and Bluetooth examples.

## Related links

- https://cafe.naver.com/geekbleofficial/14074

### CODE REVIEW

File: variants/Geekble_ESP32C3/pins_arduino.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+#include <stdint.h>
+
+#define EXTERNAL_NUM_INTERRUPTS 22
+#define NUM_DIGITAL_PINS        22
+#define NUM_ANALOG_INPUTS       6
+
+#define analogInputToDigitalPin(p)  (((p)<NUM_ANALOG_INPUTS)?(analogChannelToDigitalPin(p)):-1)
+#define digitalPinToInterrupt(p)    (((p)<NUM_DIGITAL_PINS)?(p):-1)
+#define digitalPinHasPWM(p)         (p < EXTERNAL_NUM_INTERRUPTS)
```
Comment (P-R-O-C-H-Y) [inline variants/Geekble_ESP32C3/pins_arduino.h:12]: Please remove this part as it is now defined globally per SoC.

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@SooDragon Small fix needed, but overall looks good :)

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Comment by me-no-dev** [comment]
@SooDragon please fix the conflict :) it happens when more than one PR exists for a new board.

**Comment by SooDragon** [comment]
@P-R-O-C-H-Y Thank you for ping :)

**Pull request merged**